addons/event/static/src/views/event_registration_kanban_controller.js

```diff
@@ -0,0 +1,58 @@
+import { kanbanView } from "@web/views/kanban/kanban_view";
+import { KanbanController } from "@web/views/kanban/kanban_controller";
+import { EventRegistrationSummaryDialog } from "@event/client_action/event_registration_summary_dialog";
+import { onMounted, onWillUnmount } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+export default class EventRegistrationKanbanController extends KanbanController {
+
+    setup() {
+        super.setup()
+        this.dialog = useService("dialog");
+        this.orm = useService("orm")
+        onMounted(this.hideSystray);
+        onWillUnmount(this.showSystray);
+    }
+
+    async openRecord(record, mode) {
+        if (!this.props.context.active_model) {
+            const barcode = record.data.barcode
+            const eventId = record.data.event_id[0]
+
+            const result = await this.orm.call("event.registration", "register_attendee", [], {
+                barcode: barcode,
+                event_id: eventId,
+            });
+
+            this.dialog.add(
+                EventRegistrationSummaryDialog,
+                {
+                    registration: result,
+                    model: this.model
+                }
+            )
+        } else {
+            return super.openRecord(record, mode);
+        }
+    }
+
+    hideSystray() {
+        this.observer = new MutationObserver((mutations) => {
+            document.querySelector('.o_menu_systray').classList.add('d-none');
+            this.observer.disconnect();
+        });
+        this.observer.observe(document.body, { childList: true, subtree: true });
+    }
+
+    showSystray() {
+        if (document.querySelector('.o_event_attendee_kanban_view')) {
+            document.querySelector('.o_menu_systray').classList.remove('d-none');
+        }
+    }
+}
```

Comment:
Why do we need a mutationObserver here to hide the systray? (and not on the show, it also seems)

Reply:
especially since we do not use it in the list view?

Reply:
We are using `mutationObserver` as `o_menu_systray` is not yet loaded in the DOM when we trying to access its classList. In list view its not needed as it'll already be present (as we came from the kanban view).